src/helpers/update-tasks.ts

```diff
@@ -1,120 +1,182 @@
+export async function updateTasks(context: Context) {
   const {
-    adapters: { supabase },
     logger,
-    config,
   } = context;
 
-  if (config.disqualification <= 0) {
-    logger.info("The unassign threshold is <= 0, won't unassign users.");
-  } else {
-    logger.info(`Passed the deadline on ${issue.url} and no activity is detected, removing assignees.`);
-    if (await removeAllAssignees(context, issue)) {
-      await supabase.issues.delete(issue.url);
-    }
+  const repos = await getWatchedRepos(context);
+
+  if (!repos?.length) {
+    logger.info("No watched repos have been found, no work to do.");
+    return false;
   }
+
+  for (const repo of repos) {
+    logger.info(`Updating reminders for ${repo.owner.login}/${repo.name}`);
+    await updateReminders(context, repo);
+  }
+
+  return true;
 }
 
-async function remindAssigneesForIssue(context: Context, issue: Database["public"]["Tables"]["issues"]["Row"]) {
+async function updateReminders(context: Context, repo: ListForOrg["data"][0]) {
   const {
-    adapters: { supabase },
-    logger,
-    config,
+    octokit
   } = context;
-  const now = DateTime.now();
-  const deadline = DateTime.fromISO(issue.deadline);
+  const issues = await octokit.paginate(octokit.rest.issues.listForRepo, {
+    owner: context.payload.repository.owner.login,
+    repo: repo.name,
+    per_page: 100,
+    state: "open",
+  }) as ListIssueForRepo[];
 
-  if (config.warning <= 0) {
-    logger.info("The reminder threshold is <= 0, won't send any reminder.");
-  } else {
-    const lastReminder = issue.last_reminder;
-    logger.info(`We are passed the deadline on ${issue.url}, should we send a reminder? ${!lastReminder}`);
-    if (!lastReminder && (await remindAssignees(context, issue))) {
-      await supabase.issues.upsert({
-        url: issue.url,
-        deadline: deadline.toJSDate(),
-        lastReminder: now.toJSDate(),
-        lastCheck: now.toJSDate(),
-      });
+  for (const issue of issues) {
+    if (issue.assignees?.length || issue.assignee) {
+      await updateReminderForIssue(context, repo, issue);
     }
   }
 }
 
-async function updateReminders(context: Context, issue: Database["public"]["Tables"]["issues"]["Row"]) {
+async function updateReminderForIssue(context: Context, repo: ListForOrg["data"][0], issue: ListIssueForRepo) {
   const {
-    adapters: { supabase },
     logger,
     config,
-    payload,
+    octokit
   } = context;
+  const comments = await octokit.paginate(octokit.rest.issues.listComments, {
+    owner: repo.owner.login,
+    repo: repo.name,
+    issue_number: issue.number,
+    per_page: 100,
+  }) as ListCommentsForIssue[];
+
+  const botComments = comments.filter((o) => o.user?.type === "Bot");
+  const taskDeadlineJsonRegex = /"taskDeadline": "([^"]*)"/g;
+  const taskAssigneesJsonRegex = /"taskAssignees": \[([^\]]*)\]/g;
+  const assignmentRegex = /Ubiquity - Assignment - start -/gi;
```

Comment:
This seems very risky if we ever change any message. Is there no better way?

Reply:
The intent behind the metadata was to be able to target at least the developer set "metadata class name" which in this case seems to be "Assignment".

Reply:
I have to test it to see the logic but is this used to updated a previous comment? Because if so, I don't think you get notifications for this.

Reply:
We can certainly do a test. Just message me on telegram when to check a test notification.

Reply:
Whenever I try to run it I get `Missing metadata from ...`
https://github.com/Meniole/user-activity-watcher/actions/runs/10504703437/job/29100696449#step:5:61

Reply:
It's not to update a previous comment. This just reads the bot comment that's posted when someone uses `/start` so we can extract the metadata from it

Reply:
Also the updated comment has not been merged yet so it's relying on V1 

https://github.com/ubiquibot/command-start-stop/pull/19/files#diff-47224fac5bab974df0f6fb9704d6fbb0f06375b3e98285797826104537bda3cc

in #15 I have split the functions up and you should not get that error anymore

- https://github.com/ubq-testing/user-activity-watcher/tree/chore/testing/src/helpers